salt/utils/json.py

```diff
@@ -16,6 +16,16 @@
 JSONEncoder = json.JSONEncoder
 
 
+class DatetimeEncoder(json.JSONEncoder):
+    """Datetime encoder to handle json serialization"""
+
+    def default(self, obj):
+        try:
+            return super().default(obj)
+        except TypeError:
+            return str(obj)
+
+
```

Comment:
This may be a bit too broad for our use case.  I can see cases where we want encoding to fail if the object is not serializable, but we should definitely pick out the `datetime` case at least.  Maybe something along the lines of this?

```suggestion
class DatetimeEncoder(json.JSONEncoder):
    """Datetime encoder to handle json serialization"""

    def default(self, obj):
        try:
            return super().default(obj)
        except TypeError:
            if isinstance(obj, datetime.datetime):
                # some logic to return ISO-8601 format of datetime, possibly using isoformat()
                return modified_obj
            raise
```

Reply:
I'll make the necessary changes and submit today. If you find it acceptable, I'll look into how to write test cases for salt